tcp/client/session.go

```diff
@@ -13,53 +13,37 @@ import (
 	"github.com/plgd-dev/go-coap/v3/message/codes"
 	"github.com/plgd-dev/go-coap/v3/message/pool"
 	coapNet "github.com/plgd-dev/go-coap/v3/net"
-	"github.com/plgd-dev/go-coap/v3/net/blockwise"
 	"github.com/plgd-dev/go-coap/v3/net/monitor/inactivity"
-	"github.com/plgd-dev/go-coap/v3/net/responsewriter"
-	"github.com/plgd-dev/go-coap/v3/options/config"
-	coapSync "github.com/plgd-dev/go-coap/v3/pkg/sync"
 	"github.com/plgd-dev/go-coap/v3/tcp/coder"
 	"go.uber.org/atomic"
 )
 
 type Session struct {
 	// This field needs to be the first in the struct to ensure proper word alignment on 32-bit platforms.
 	// See: https://golang.org/pkg/sync/atomic/#pkg-note-BUG
-	sequence                        atomic.Uint64
-	onClose                         []EventFunc
-	ctx                             atomic.Value // // TODO: change to atomic.Pointer[context.Context] for go1.19
-	inactivityMonitor               InactivityMonitor
-	errSendCSM                      error
-	cancel                          context.CancelFunc
-	done                            chan struct{}
-	goPool                          GoPoolFunc
-	errors                          ErrorFunc
-	blockWise                       *blockwise.BlockWise[*Conn]
-	connection                      *coapNet.Conn
-	handler                         func(*responsewriter.ResponseWriter[*Conn], *pool.Message)
-	tokenHandlerContainer           *coapSync.Map[uint64, func(*responsewriter.ResponseWriter[*Conn], *pool.Message)]
-	messagePool                     *pool.Pool
-	mutex                           sync.Mutex
-	maxMessageSize                  uint32
-	peerBlockWiseTranferEnabled     atomic.Bool
-	peerMaxMessageSize              atomic.Uint32
-	connectionCacheSize             uint16
-	disableTCPSignalMessageCSM      bool
-	disablePeerTCPSignalMessageCSMs bool
-	blockwiseSZX                    blockwise.SZX
-	closeSocket                     bool
+	sequence          atomic.Uint64
+	onClose           []EventFunc
+	ctx               atomic.Value // // TODO: change to atomic.Pointer[context.Context] for go1.19
+	inactivityMonitor InactivityMonitor
+	errSendCSM        error
+	cancel            context.CancelFunc
+	done              chan struct{}
+	errors            ErrorFunc
+	connection        *coapNet.Conn
+	messagePool       *pool.Pool
+	mutex             sync.Mutex
```

Comment:
```
private struct {
  mutex sync.Mutex
  onClose []EventFunc
  ctx               atomic.Value // TODO: change to atomic.Pointer[context.Context] for go1.19
}
```

Though since `ctx` is atomic and contexts have value semantics, it shouldn't need to be protected by a mutex.

Reply:
Fixed